client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogma.java

```diff
@@ -343,7 +364,8 @@
      *         {@link PathPattern} in the specified repository
      */
     CompletableFuture<List<Commit>> getHistory(
-            String projectName, String repositoryName, Revision from, Revision to, PathPattern pathPattern);
```

Comment:
I just realized with the current pattern, we will need to make a breaking change every time a new parameter is added 🤔 
I wonder if we should add `@UnstableApi` to methods of this sort in this class

Reply:
I am also worried about that. I think we have to design a new API to add new options without deprecation or overloaded methods.
How about designing a new API with command and interpreter pattern. That might be pretty similar to the server-side command pattern.
```java

public interface CentralDogma {
    <T> CompletableFuture<T> execute(CentralDogmaRequest<T> request);
}

public interface CentralDogmaRequest<T> {
}

public class CreateRepository implements CentralDogmaRequest<CentralDogmaRepository> {
    private String projectName;
    private String repositoryName;
    
    // constructor and getters
    ...
}

public class HistoryRequest implements CentralDogmaRequest<List<Commit>> {
    ...

    public CompletableFuture<List<Commit>> get(Revision from, Revision to) {
        this.from = from;
        this.to = to;

        return centralDogmaRepo.centralDogma().execute(this);
    }
}
```

Reply:
That's a good suggestion. 👍 Let me use the command pattern. 😄

Reply:
Had a chat with @ikhoon and @jrhee17, and we decide to keep the current style because `CentralDogma` client plays a role of API spec.
However, we might reconsider thie decision later if we need to add more parameter frequently or we add Armeria dependency to `centraldogma-client` module. #673